File: 7-kyu/clothes-size-number-converter/javascript/solution.js
const sizeToNumber = size => {
  if (!(/^x*[sl]$/g.test(size) || /^m$/g.test(size))) {
    return null;
  }
  const xModifier = 2;
  const xCount = [...size].filter((ch) => ch === "x").length * xModifier;
  const mediumBase = 38;
  const smallBase = 36;
  const largeBase = 40;
  if (size === "m") {
    return mediumBase;
  } else if (size.includes("s")) {
    return smallBase - xCount;
  }
  return largeBase + xCount;
};

File: 7-kyu/clothes-size-number-converter/python/solution.py
import re

def size_to_number(size):
    if size == 'm':
        return 38
    if not (re.match(r'^x*[sl]$', size) or re.match(r'^m$', size)):
        return None
    x_count = size.count('x')
    bases = {'s': 36, 'l': 40}
    return bases['s'] - x_count * 2 if 's' in size else bases['l'] + x_count * 2
